lua/bufferline/options.lua

```diff
@@ -42,7 +51,82 @@ local DEFAULT_DIAGNOSTICS = {
 --- @field inactive? boolean
 --- @field visible? boolean
 
---- @alias bufferline.options.icons boolean|"both"|"buffer_number_with_icon"|"buffer_numbers"|"numbers"
+--- @class bufferline.options.icons.filetype
+--- @field custom_color? boolean if present, this color will be used for ALL filetype icons
+--- @field enabled? boolean iff `true`, show the `devicons` for the associated buffer's `filetype`.
+
+--- @class bufferline.options.icons.buffer
+--- @field buffer_index? boolean iff `true`, show the index of the associated buffer with respect to the ordering of the buffers in the tabline.
+--- @field buffer_number? boolean iff `true`, show the `bufnr` for the associated buffer.
+--- @field button? string the button which is clicked to close / save a buffer, or indicate that it is pinned.
+--- @field filetype? bufferline.options.icons.filetype filetype icon options
+--- @field separator? string the left-hand separator between buffers in the tabline
+
+--- @class bufferline.options.icons.state: bufferline.options.icons.buffer
+--- @field modified? bufferline.options.icons.buffer the icons used for an modified buffer
+--- @field pinned? bufferline.options.icons.buffer the icons used for a pinned buffer
+
+--- @class bufferline.options.icons: bufferline.options.icons.state
+--- @field alternate? bufferline.options.icons.state the icons used for an alternate buffer
+--- @field current? bufferline.options.icons.state the icons for the current buffer
+--- @field inactive? bufferline.options.icons.state the icons for inactive buffers
+--- @field visible? bufferline.options.icons.state the icons for visible buffers
+
+--- @alias bufferline.options.icons.preset boolean|"both"|"buffer_number_with_icon"|"buffer_numbers"|"numbers"
+
+--- @type {[bufferline.options.icons.preset]: bufferline.options.icons}
+local PRESETS = {
```

Comment:
This is to translate the old `icons = true|false|'both'|…` scheme automatically.